### ISSUE TITLE
"No cookie found!" in Console

### ISSUE THREAD

**Issue by gian-reto** [supabase-community/auth-helpers#114]
My Next app has a mix of public and protected pages. As shown in the example, I wrapped all pages in a `UserProvider`:

```javascript
import type { AppProps } from "next/app";
import { UserProvider } from "@supabase/supabase-auth-helpers/react";
import { supabaseClient } from "@supabase/supabase-auth-helpers/nextjs";

function App({ Component, pageProps }: AppProps) {
  return (
    <UserProvider supabaseClient={supabaseClient}>
      <Component {...pageProps} />
    </UserProvider>
  );
}

export default App;
```

However, on all pages, the error message `No cookie found!` is being printed to console if the user is not logged in. I believe this happens because the `UserProvider` is requesting `/api/auth/user`, which is handled by [getUser.ts](https://github.com/supabase-community/supabase-auth-helpers/blob/main/src/nextjs/utils/getUser.ts). `getUser.ts` checks if an `access_token` is present, which of course is always false if the user is not logged in. 

The problem is that this happens even on pages that don't even need the user to be logged in, e.g. my home page:

```javascript
import Head from "next/head";
import type { NextPage } from "next";

const HomePage: NextPage = () => {
  return (
    <div>
      <Head>
        <title>Hello world!</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main>
        <h1>Hello world!</h1>
      </main>
    </div>
  );
};

export default HomePage;
```

Is this the desired behavior or is there a way to only request the user if a page uses `useUser()`? I tried adding `export const getServerSideProps = withPageAuth({ authRequired: false });` to the public pages, but doesn't change anything.

Thanks for your help! 🎉

**Comment by samtietjen** [comment]
Also experiencing this.

**Comment by nahtnam** [comment]
I'm also experiencing this with `useUser` and this error is printed by the hook `Error: The request to /api/auth/user failed`, but upon looking at the network inspector, I see that the `/api/auth/user` endpoint returns the no cookie found error

**Comment by silentworks** [comment]
We will be addressing this in our next release.

**Comment by adamk22** [comment]
Same issue here and have the same situation as OP with mixed protected and public pages and using the `UserProvider`. When making calls from the `getServerSideProps` function I'm getting a JWT expired error even though the RLS for the `SELECT` is set to allow for all users. 

So:

```
export async function getServerSideProps() {
	const items = await getItems();
	console.log("items: ", items);
	return {
		props: {
			items: items.data,
		},
	};
}

export default function Home(props: ServerProps) {
...
}

```

Response logged:
```
{
  error: { message: 'JWT expired' },
  data: null,
  count: null,
  status: 401,
  statusText: 'Unauthorized',
  body: null
}
```
And also have the `No cookie found!` error in the console.

**Comment by pdhoward** [comment]
same +1

**Comment by samtietjen** [comment]
Anyone find a work around for this?

**Comment by silentworks** [comment]
@samtietjen for now you can place an if statement to check for that error and not display it. The error is currently being caused by a bug #134  which I will try and address next week. We are also trying to come up with proper error codes and a glossy.

**Comment by lachlanhawthorne** [comment]
Here's a work around to check if the cookie exists before returning handleAuth() in `/pages/api/auth/[...supabase].ts`

```tsx
import { handleAuth } from "@supabase/supabase-auth-helpers/nextjs"
import type { NextApiRequest, NextApiResponse } from "next"

export default async function AuthHandler(req: NextApiRequest, res: NextApiResponse) {
  return req.headers.cookie 
    ? handleAuth() 
    : res.status(200).send('ok')
}
```

And for getServerSideProps
```tsx
import type { GetServerSidePropsContext } from 'next'

export async function getServerSideProps(context: GetServerSidePropsContext) {
  if(context.req.headers.cookie) {
    const user = getUser(context)
    // server side code
  }
}
```

**Comment by lachlanhawthorne** [comment]
Here's a work around to check if the cookie exists or route is '/api/auth/callback' before returning handleAuth() in `/pages/api/auth/[...supabase].ts`

```tsx
import { handleAuth } from "@supabase/supabase-auth-helpers/nextjs"
import { NextApiRequest, NextApiResponse } from "next"

export default function AuthHandler(req: NextApiRequest, res: NextApiResponse) {
  if(req.headers.cookie || req.query.supabase[0] === 'callback') {
    handleAuth()(req, res)
  } else {
    res.status(200).json({})
  } 
}
```

And for getServerSideProps
```tsx
import type { GetServerSidePropsContext } from 'next'

export async function getServerSideProps(context: GetServerSidePropsContext) {
  if(context.req.headers.cookie) {
    const user = getUser(context)
    // server side code
  }
}
```

**Comment by travispamaral** [comment]
I'm actually only seeing this after calling `signIn({provider: 'google'})`. Is it related to the same bug? Only happens to be when I fire that call. Nothing actually happens and I need to click the button a few times 🤔

**Comment by silentworks** [comment]
I've just released `@supabase/auth-helpers-nextjs@0.2.1` which should have a fix for this issue.
